### ISSUE TITLE
mesa doesn't provide a pkgconfig file for libGL?

### ISSUE THREAD

**Issue by Begasus** [haikuports/haikuports#6514]
Trying to rebuild libreoffice because a dependency for libepoxy can't be found.
libGL is in the recipe for libepoxy, but running a build shows that even libepoxy can't find libgl (finds libegl fine), reading through the mesa log I found out that pkg-config fails to find it and when I look into the mesa-devel package I don't see a gl.pc file in there.

**Comment by Begasus** [comment]
Maybe it just doesn't provide one: https://github.com/anholt/libepoxy/pull/156
Removing the libGL dependency for libepoxy seems to work ok for libreoffice.

**Comment by Begasus** [comment]
Seems to be invoked here? https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/meson.build#L140

**Comment by kallisti5** [comment]
I commented on this one in irc.

So, gl.pc seems to indicate an expectation that pkg-config will provide the needed libraries for libGL.

*however* on linux libGL is actually glx (C OpenGL init code).   On Haiku libGL is actually hgl (our C++, BGLView API).  glx is  Linuxism (that BSD also uses since they mimic Linux's DRM/DRI/GLX)

That raises an interesting question.  If we offered a gl.pc, we would be offering a different API than what anything seeking gl.pc would expect (still breaking the thing in question's build)

On Linux, gl.pc seems to offer the following:

```
$ pkg-config gl --libs --cflags
-lGL 
```

That's not a whole lot.


So.. the question becomes:

1) Should we offer a gl.pc which offers up a different API than what the program is expecting (needing to add HAIKU ifdef's in C/C++ code)
2) Should we continue to omit gl.pc and ensure the applications have Haiku support in their built system, and in their code (needing to add HAIKU Ifdef's in C/C++ code)

I think #2 is the better solution, since it will stop porters before getting too far in the port that dragons are ahead.    #1 might create an issue where the build system of an application "thinks" it has GLX, but doesn't (creating more headaches for porters)


tldr; from my analysis, I think that offering gl.pc provides "intent" to applications that you have a glx libGL (which we don't)

**Comment by kallisti5** [comment]
extending a little bit more.  If you're porting applications to Haiku, you can assume -lGL is always present.

the haiku package (due to GLTeapot, Haiku3D, etc) depends on mesa.  Our OpenGL kit is also assumed to always exist (even though I outsourced the code from src/kits/opengl to mesa)

You would be just as valid detecting haiku, and always assuming libGL should be present (if it isn't an optional piece of an application)

**Comment by kallisti5** [comment]
@waddlesplash this all opens the horrible idea in my brain of offering hgl *and* glx api's in our libGL.

We could translate glx calls to hgl calls somehow (likely after getting a drm/drm compatibility layer) and do everything Linux does from an API standpoint... but with more flare :tada:

**Comment by pulkomandy** [comment]
Previously we had implelented egl which is a more standardized API for this (specified by Khronos Group like OpenGL). It is used in embedded Linux when you want to do OpenGL or GLES without starting X11 (directly over framebuffer). 

This has reasonably good support (for example in SDL2, libepoxy, ...) which means most apps would be happy with that (few of them actually use glx directly, because that would mean different code for Linux, Windows and MacOS). In fact we run into this type of problem only when porting one of these abstraction layers.

However, our implementation was not maintained in newer Mesa versions and is now crashing.

**Comment by kallisti5** [comment]
yeah.. EGL is indeed a better choice.   we likely should focus on fixing our mesa libEGL library.

That doesn't solve the issue with libreoffice above, but it the better path.

**Comment by pulkomandy** [comment]
Libreoffice just uses libepoxy. It is not clear where the issue is exactly. I'm not sure how libepoxy was ported exactly, but it appears it hardcodes usage of pkgconfig for finding libGL in its meson buildfiles. I'm not sure because I didn't see a build log with the exact problem? (I guess there is one now, but I didn't watch the IRC channel for the last few days)

**Comment by Begasus** [comment]
> I'm not sure because I didn't see a build log with the exact problem?

For what it's worth, in the last run(s) I did for libreoffice I didn't see the problem re-appear, but have been side tracked in the meantime.

**Comment by waddlesplash** [comment]
I patched libepoxy. I don't think we need to do anything else for Mesa here in particular.
